src/main/java/net/minecraftforge/java_provisioner/Disco.java

```diff
@@ -623,72 +713,31 @@ public static Arch byKey(String key) {
             return null;
         }
 
-        private static Arch getCurrent() {
-            String prop = System.getProperty("os.arch").toLowerCase(Locale.ENGLISH);
-            for (Arch value : $values) {
-                for (String name : value.names) {
-                    if (prop.equals(name)) {
-                        return value;
+        private static final class CurrentArch {
+            private static final Arch CURRENT;
+
+            static {
+                Arch current = UNKNOWN;
```

Comment:
Not sure what the benefit of moving this to a static bloc instead of a normal function is.. you could of just done `private static final Arch CURRENT = getCurrent()`